File: Easy/JavaScript/2103. Rings and Rods.js
var countPoints = function(rings) {
    let count = 0;
    let red = Array(10).fill(0);
    let blue = Array(10).fill(0);
    let green = Array(10).fill(0);

    // Even position of rings
    for(let i = 0; i < rings.length; i += 2){
        let c = rings.charAt(i);
        let index = rings.charAt(i + 1) - '0';
        if(c == 'R')
            red[index]++;
        if(c == 'B')
            blue[index]++;
        if(c == 'G')
            green[index]++;
    }

    // Odd position of rings
    for(let i = 0; i < 10; i++){
        if(red[i] != 0 && blue[i] != 0 && green[i] != 0)
            count++;
    }

    return count;
};

File: Easy/Python/2103. Rings and Rods.py
class Solution(object):
    def countPoints(self, rings):
        count = 0
        red = [0] * 10
        blue = [0] * 10
        green = [0] * 10

        # Even position of rings
        for i in range(0, len(rings), 2):
            c = rings[i]
            index = int(rings[i + 1])
            if c == 'R':
                red[index] += 1
            if c == 'B':
                blue[index] += 1
            if c == 'G':
                green[index] += 1

        # Odd position of rings
        for i in range(10):
            if red[i] != 0 and blue[i] != 0 and green[i] != 0:
                count += 1

        return count
